### PR TITLE
Execaction in rem.auth

### PR DESCRIPTION
**Pull request by Alladin9393**
## Change Description

Implement `execaction` in `rem.auth` contract to execute custom actions without authorization and use `rem.auth` as proxy. 

## Implemented
- Create `execaction`
- Add `rem.auth` tests

### CODE REVIEW

File: contracts/rem.auth/src/rem.auth.cpp [hunk 1]
```diff
@@ -67,6 +67,49 @@ namespace eosio {
       cleanupkeys();
    }
 
+   void auth::execaction(const name &account, action& act, const block_timestamp &action_timestamp,
+                         const public_key &pub_key, const signature &signed_by_pub_key, const name &payer)
```
Comment (roman-tik) [inline contracts/rem.auth/src/rem.auth.cpp]: **signed_by_pub_key** rename to **action_data_signature**
// we do not sign by a public key :)

File: contracts/rem.auth/src/rem.auth.cpp [hunk 2]
```diff
@@ -67,6 +67,49 @@ namespace eosio {
...
+   {
+      bool is_payer = bool(payer);
+      name payer_name = is_payer ? payer : account;
+      if (is_payer) { require_auth(payer_name); }
```
Comment (roman-tik) [inline contracts/rem.auth/src/rem.auth.cpp]: why do we need **payer** in this action?

File: tests/rem_auth_tests.cpp [hunk 3]
```diff
@@ -1781,4 +1814,84 @@ BOOST_FIXTURE_TEST_CASE( keys_cleanup_test, rem_auth_tester ) {
...
+      crypto::private_key new_key_priv = crypto::private_key::generate();
+      crypto::public_key new_key_pub   = key_priv.get_public_key();
+      auto signed_by_new_key_execaction = new_key_priv.sign(digest_execaction);
+      // Error expected key different than recovered key
+      BOOST_REQUIRE_THROW(execaction(account, act, action_timestamp, key_pub, signed_by_new_key_execaction, payer, auths_level),
+                          crypto_api_exception);
+      // account has no linked application keys
+      BOOST_REQUIRE_THROW(execaction(account, act, action_timestamp, new_key_pub, signed_by_new_key_execaction, payer, auths_level),
+                        crypto_api_exception);
+   } FC_LOG_AND_RETHROW()
```
Comment (roman-tik) [inline tests/rem_auth_tests.cpp:1890]: add a test case where **proda** correctly calls **execaction** but the nested action transfers tokens from **prodb** to **prodc**

**Review by roman-tik — CHANGES_REQUESTED** [review]

File: contracts/rem.auth/src/rem.auth.cpp (hunk 1 above)
Comment (Alladin9393) [reply contracts/rem.auth/src/rem.auth.cpp]: @roman-tik, done.

File: tests/rem_auth_tests.cpp (hunk 3 above)
Comment (Alladin9393) [reply tests/rem_auth_tests.cpp:1890]: @roman-tik, done.

File: contracts/rem.auth/src/rem.auth.cpp (hunk 2 above)
Comment (Alladin9393) [reply contracts/rem.auth/src/rem.auth.cpp]: @roman-tik, done.

**Review by roman-tik — APPROVED** [review]

**Pull request merged**